### PR TITLE
#22-Configuration Service added.

### PR DESCRIPTION
**Pull request by morteza-azizi**

### CODE REVIEW

File: business/services/ConfigurationService.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/decentralized-cloud/Tenant/business/contracts"
+)
+
+// ConfigurationService implements the service that provides environment variables
+type ConfigurationService struct {
+}
+
+// NewConfigurationService creates new instance of the ConfigurationService, setting up all dependencies and returns the instance
+// Returns the new service or error if something goes wrong
+func NewConfigurationService(contracts.ConfigurationServiceContract, error) {
```
Comment (mortezaalizadeh) [inline business/services/ConfigurationService.go]: I think you have the argument inside out, aren't they supposed to be the return arguments?

File: business/contracts/ConfigurationServiceContract.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Package contracts defines the configuration business contracts
```
Comment (mortezaalizadeh) [inline business/contracts/ConfigurationServiceContract.go]: "Configuration business" contract?

File: go.mod [hunk 3]
```diff
@@ -4,12 +4,14 @@ go 1.13
 
 require (
 	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a // indirect
+	github.com/craicoverflow/go-environment-variables-example v0.0.0-20190228224948-ee3fabeff275
```
Comment (mortezaalizadeh) [inline go.mod]: What is the reason behind adding this dependency here?

File: go.mod [hunk 4]
```diff
@@ -4,12 +4,14 @@ go 1.13
...
 	github.com/decentralized-cloud/TenantContract v0.0.2
 	github.com/go-kit/kit v0.9.0
 	github.com/go-logfmt/logfmt v0.4.0 // indirect
 	github.com/go-ozzo/ozzo-validation v3.6.0+incompatible
 	github.com/go-stack/stack v1.8.0 // indirect
 	github.com/golang/mock v1.3.1
+	github.com/joho/godotenv v1.3.0
```
Comment (mortezaalizadeh) [inline go.mod]: Same here, what is the reason behind adding this dependency here?

File: business/services/ConfigurationService.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package services
```
Comment (mortezaalizadeh) [inline business/services/ConfigurationService.go]: The package comment is missing, you will get go-lint warning later

**Review by mortezaalizadeh — CHANGES_REQUESTED** [review]
The configuration service and contract files need to be moved into their own directory and have their own package. They will be used later everywhere that requires to access the configuration

File: business/services/ConfigurationService.go (hunk 5 above)
Comment (morteza-azizi) [reply business/services/ConfigurationService.go]: fix applied.

File: go.mod (hunk 4 above)
Comment (morteza-azizi) [reply go.mod]: fix applied.

File: go.mod (hunk 3 above)
Comment (morteza-azizi) [reply go.mod]: fix applied.

File: business/contracts/ConfigurationServiceContract.go (hunk 2 above)
Comment (morteza-azizi) [reply business/contracts/ConfigurationServiceContract.go]: fix applied.

File: business/services/ConfigurationService.go (hunk 1 above)
Comment (morteza-azizi) [reply business/services/ConfigurationService.go]: not sure if fix applied is correct.

**Review by mortezaalizadeh — APPROVED** [review]

**Pull request merged**